File: 02.Databases_Fundamentals_2016/02.Databases_Advanced_Hibernate/05.HibernateRelations/02.StudentSystem/student-system/src/main/java/app/domain/Resource.java
package app.domain;

import app.enums.ResourceType;

import javax.persistence.*;

@Entity
@Table(name = "resources")
public class Resource {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "resource_id")
    private Long resourceId;

    @Basic
    private String name;

    @Enumerated(value = EnumType.STRING)
    @Column(name = "resource_type")
    private ResourceType resourceType;

    @Basic
    private String url;

    @ManyToOne
    @JoinColumn(name = "course_id", referencedColumnName = "course_id")
    private Course course;

    public Resource() {
    }

    public Long getResource_id() {
        return resourceId;
    }

    public void setResource_id(Long resourceId) {
        this.resourceId = resourceId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public ResourceType getResourceType() {
        return resourceType;
    }

    public void setResourceType(ResourceType resourceType) {
        this.resourceType = resourceType;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public Course getCourse() {
        return course;
    }

    public void setCourse(Course course) {
        this.course = course;
    }
}

File: 00.My_Practice_Courses/07.C#_Web_Development_Basics/01.Introdution_.NET_Core/02.Exercises/StudentSystem/StudentSystemData/Models/Resource.cs
﻿using StudentSystemData.Models.Enums;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace StudentSystemData.Models
{
    public class Resource
    {
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string Url { get; set; }

        public ResourceType ResourceType { get; set; }

        public int CourseId { get; set; }

        public Course Course { get; set; }

        public List<License> Licenses { get; set; } = new List<License>();
    }
}
